bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/CommonInterface.java

```diff
@@ -121,13 +121,10 @@ default Stream<Channel> getActiveChannels() {
                 .filter(channel -> ChannelKind.STATE.equals(channel.getKind()) && isLinked(channel.getUID()));
     }
 
-    default Optional<CommonInterface> getHomeHandler() {
-        CommonInterface bridgeHandler = getBridgeHandler();
-        if (bridgeHandler != null) {
-            return bridgeHandler.getCapabilities().get(HomeCapability.class).isPresent() ? Optional.of(bridgeHandler)
-                    : Optional.empty();
-        }
-        return Optional.empty();
+    default Optional<CommonInterface> recurseUpToHomeHandler(@Nullable CommonInterface handler) {
+        return handler == null ? Optional.empty()
+                : handler.getCapabilities().get(HomeCapability.class).isPresent() ? Optional.of(handler)
+                        : recurseUpToHomeHandler(handler.getBridgeHandler());
```

Comment:
Chaining the ternary operator results in code that is hard to read and understand. Better use two lines. 

```suggestion
    default Optional<CommonInterface> recurseUpToHomeHandler(@Nullable CommonInterface handler) {
        if (handler == null) {
            return Optional.empty();
        }
        return handler.getCapabilities().get(HomeCapability.class).isPresent() ? Optional.of(handler)
                        : recurseUpToHomeHandler(handler.getBridgeHandler());
```

```suggestion
    default Optional<CommonInterface> recurseUpToHomeHandler(@Nullable CommonInterface handler) {
        return handler == null ? Optional.empty()
                : handler.getCapabilities().get(HomeCapability.class).isPresent() ? Optional.of(handler)
                        : recurseUpToHomeHandler(handler.getBridgeHandler());
```

Reply:
OK, changed accordingly.